deepsubtree.go

```diff
@@ -0,0 +1,381 @@
+package iavl
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"strings"
+
+	ics23 "github.com/confio/ics23/go"
+)
+
+// Represents a IAVL Deep Subtree that can contain
+// a subset of nodes of an IAVL tree
+type DeepSubTree struct {
+	*MutableTree
+}
+
+// Traverses in the nodes in the NodeDB in the Deep Subtree
```

Comment:
[non-blocking] 
If accurate, I think this is a little clearer. Too many `in`s before 
```suggestion
// Traverses the nodes in the NodeDB that are part of Deep Subtree
```